Modules/atexitmodule.c

```diff
@@ -118,10 +119,14 @@ _PyAtExit_Fini(PyInterpreterState *interp)
 
 
 static void
-atexit_callfuncs(struct atexit_state *state)
+atexit_callfuncs(struct atexit_state *state, int from_python_module)
 {
     assert(!PyErr_Occurred());
 
+    if (!from_python_module) {
```

Comment:
because Lib/_test_atexit.py calls `atexit._run_callbacks()` which shouldn't set the flag.